modules/services/window-managers/i3-sway/lib/functions.nix

```diff
@@ -77,88 +77,67 @@ rec {
-            optionalString (colors.separator != null)
-            "separator ${colors.separator}"
-          }
-          ${
-            optionalString (colors.focusedBackground != null)
-            "focused_background ${colors.focusedBackground}"
-          }
-          ${
-            optionalString (colors.focusedStatusline != null)
-            "focused_statusline ${colors.focusedStatusline}"
-          }
-          ${
-            optionalString (colors.focusedSeparator != null)
-            "focused_separator ${colors.focusedSeparator}"
-          }
-          ${
-            optionalString (colors.focusedWorkspace != null)
-            "focused_workspace ${barColorSetStr colors.focusedWorkspace}"
-          }
-          ${
-            optionalString (colors.activeWorkspace != null)
-            "active_workspace ${barColorSetStr colors.activeWorkspace}"
-          }
-          ${
-            optionalString (colors.inactiveWorkspace != null)
-            "inactive_workspace ${barColorSetStr colors.inactiveWorkspace}"
-          }
-          ${
-            optionalString (colors.urgentWorkspace != null)
-            "urgent_workspace ${barColorSetStr colors.urgentWorkspace}"
-          }
-          ${
-            optionalString (colors.bindingMode != null)
-            "binding_mode ${barColorSetStr colors.bindingMode}"
-          }
-        ${optionalString colorsNotNull "}"}
-        ${extraConfig}
       }
     '';
 
   gapsStr = with cfg.config.gaps; ''
-    ${optionalString (inner != null) "gaps inner ${toString inner}"}
-    ${optionalString (outer != null) "gaps outer ${toString outer}"}
-    ${optionalString (horizontal != null)
-    "gaps horizontal ${toString horizontal}"}
-    ${optionalString (vertical != null) "gaps vertical ${toString vertical}"}
-    ${optionalString (top != null) "gaps top ${toString top}"}
-    ${optionalString (bottom != null) "gaps bottom ${toString bottom}"}
-    ${optionalString (left != null) "gaps left ${toString left}"}
-    ${optionalString (right != null) "gaps right ${toString right}"}
-
-    ${optionalString smartGaps "smart_gaps on"}
-    ${optionalString (smartBorders != "off") "smart_borders ${smartBorders}"}
+    ${concatStringsSep "\n" (notEmpty [
+      (optionalString (inner != null) "gaps inner ${toString inner}")
+      (optionalString (outer != null) "gaps outer ${toString outer}")
+      (optionalString (horizontal != null)
+        "gaps horizontal ${toString horizontal}")
+      (optionalString (vertical != null) "gaps vertical ${toString vertical}")
+      (optionalString (top != null) "gaps top ${toString top}")
+      (optionalString (bottom != null) "gaps bottom ${toString bottom}")
+      (optionalString (left != null) "gaps left ${toString left}")
+      (optionalString (right != null) "gaps right ${toString right}")
+      (optionalString smartGaps "smart_gaps on")
+      (optionalString (smartBorders != "off") "smart_borders ${smartBorders}")
+    ])}
   '';
```

Comment:
Can you remove the outer string? It's not needed anymore.

Reply:
With outer string you mean `${concatStringsSep "\n" (notEmpty [`?

Reply:
Why are they not needed anymore? I can't find a drop notice/some explanation at i3-gaps...